### ISSUE TITLE
Skill dropdown crash

### ISSUE THREAD

**Issue by Nightblade** [PathOfBuildingCommunity/PathOfBuilding#6040]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### Screenshots of crash/graphical issue, version and options menu

![image](https://user-images.githubusercontent.com/23911/231705946-ee6ccb84-b5cb-4429-96d4-a93f3713e156.png)

![image](https://user-images.githubusercontent.com/23911/231712259-bcaf4744-e78a-4549-870f-37991107c01d.png)

### What are your system specifications and configuration?

Win10

### How to reproduce the issue

1. Click on the `Vaal Ice Shot` group (in the Skills tab)
2. `ctrl+c` & `ctrl+v` to duplicate the group,
4. Click `Include in Full DPS` on the new group,
5. Click on the old group, 
6. Click on the empty skill slot (below Added Cold Damage),
7. Error popup 🥓 

### Character build code

```shell
https://pobb.in/jXCrnzwUSVBG
```

**Comment by RecurringNightmare** [comment]
yes, same issue here

the bug actually has a history
in the last days i noticed that the damage calculation for my selected main skill would confuse my first galvanic arrow skill group for self-use (trinity/mirage archer) and my second skill group that also has galvanic arrow but this time with ballista support
the drop-down would still show both skill groups and let me select them and if i wanted arrow or cone calculation, but both calculated for the totems and not for me attacking myself
deleting all skill groups and starting fresh solved that
now this new issue happens and it also has to do with two skill groups having the same skill gem and the main skill selection confusing them, apparently entering a loop and creating an overflow

in other words, if you have two skill groups that have the same skill gem and one of them is your selected main skill on the top left, trying to get the support gem dropdown list to show will likely produce the crash, although i just managed to get it to work normally for a minute, just to have the crash then happen again

if you make a totally different skill the main skill you can still work on your build, full dps etc still get calculated normally on the bottom left, just the details for the main skill on top left obviously are missing/wrong then...

hope they get it fixed soon, really annoying...

**Comment by Peechey** [comment]
Looks like this is caused by something introduced in 2.28.0 as I'm not able to reproduce the crash using these steps on 2.27.2

**Comment by Paliak** [comment]
It's acaused by @Nostrademous 's implementation of manaforged arrows. Fix on the way.
